main.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/DataDog/datadog-operator/controllers"
+	"github.com/DataDog/datadog-operator/pkg/controller/debug"
+	"github.com/DataDog/datadog-operator/pkg/version"
+	edsdatadoghqv1alpha1 "github.com/DataDog/extendeddaemonset/api/v1alpha1"
+	"github.com/DataDog/extendeddaemonset/pkg/controller/metrics"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	// +kubebuilder:scaffold:imports
+)
+
+const (
+	maximumGoroutines = 200
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(apiregistrationv1.AddToScheme(scheme))
+
+	utilruntime.Must(datadoghqv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(edsdatadoghqv1alpha1.AddToScheme(scheme))
+	// +kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "enable-leader-election", true,
+		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
+
+	// Custom flags
+	var printVersion, pprofActive, supportExtendedDaemonset bool
+	logLevel := zap.LevelFlag("loglevel", zapcore.InfoLevel, "Set log level")
+	flag.BoolVar(&printVersion, "version", false, "Print version and exit")
+	flag.BoolVar(&pprofActive, "pprof", false, "Enable pprof endpoint")
+	flag.BoolVar(&supportExtendedDaemonset, "supportExtendedDaemonset", false, "Support usage of Datadog ExtendedDaemonset CRD.")
+	flag.Parse()
+
+	// Parsing flags
+	flag.Parse()
+
+	// Logging setup
+	customSetupLogging(*logLevel)
+
+	// Print version information
+	if printVersion {
+		version.PrintVersionWriter(os.Stdout, "text")
+		os.Exit(0)
+	}
+
+	version.PrintVersionLogs(setupLog)
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		HealthProbeBindAddress: ":8081",
+		Port:                   9443,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "c4f7cd04.datadoghq.com",
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	// Custom setup
+	customSetupHealthChecks(mgr)
+	customSetupEndpoints(pprofActive, mgr)
+
+	// Get some information about Kubernetes version
+	if err := controllers.SetupControllers(mgr, supportExtendedDaemonset); err != nil {
+		setupLog.Error(err, "unable to start controllers")
+		os.Exit(1)
+	}
+
+	// +kubebuilder:scaffold:builder
+
+	setupLog.Info("starting manager")
+	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+		setupLog.Error(err, "problem running manager")
+		os.Exit(1)
+	}
+}
+
+func customSetupLogging(logLevel zapcore.Level) {
+	encoderConfig := zap.NewProductionEncoderConfig()
+	encoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
```

Comment:
could you add the support of JSON log (like for [EDS](https://github.com/DataDog/extendeddaemonset/pull/48)) 
because the console Format is not properly understood by our default parser.